app/models/coupon.rb

```diff
@@ -1,4 +1,4 @@
 class Coupon < ApplicationRecord
   validates :amount, :name, presence: true
-  has_many :order_items, as: :source
+  has_many :order_items, as: :source, dependent: :destroy
```

Comment:
@fiestacasey should this be on the product and order_item? i know yesterday you said product, but I think it would make sense on the order_item as well.